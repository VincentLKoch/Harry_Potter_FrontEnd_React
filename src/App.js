import React from "react";
import { connect } from "react-redux";
import HouseBanner from "./Components/HouseBanner";

const App = props => {
  return (
    <div className="App">
      a
      <br />
      <HouseBanner id={0} />
      <br />
      <HouseBanner id={1} />
      <br />
      <HouseBanner id={2} />
      <br />
      <HouseBanner id={3} />
    </div>
  );
};

const mapStateToProps = state => {
  return {
    houses: state.houses
  };
};

const mapDispatchToProps = dispatch => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
