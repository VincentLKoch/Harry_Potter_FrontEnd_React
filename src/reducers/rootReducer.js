import { combineReducers } from "redux";
import housesReducer from "./housesReducer";
import waitingReducer from "./waitingReducer";
import wizardsReducer from "./wizardsReducer";

export default combineReducers({
  houses: housesReducer,
  wizards: wizardsReducer,
  wait: waitingReducer
});
