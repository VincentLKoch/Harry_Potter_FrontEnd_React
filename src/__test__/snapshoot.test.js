import React from "react";
import renderer from "react-test-renderer";

import App from "../App";

import { createStore } from "redux";
import { Provider, connect } from "react-redux";
import rootReducer from "../reducers/rootReducer";

const store = createStore(rootReducer);

describe("Testing Snapshoot:", () => {
  let testRenderer;
  let testInstance;
  it("Render correctly", () => {
    try {
      testRenderer = renderer.create(
        <Provider store={store}>
          <App />
        </Provider>
      );
      testInstance = testRenderer.root;
    } catch (error) {
      expect(error).toBeNull();
    }

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it("test images:", () => {
    const images = testInstance.findAllByType("img");
    expect(images.length).toBe(4);

    expect(images[0].props).toEqual({
      src: "./images/Gryffindor.png",
      alt: "Gryffindor"
    });
    expect(images[1].props).toEqual({
      src: "./images/Hufflepuff.png",
      alt: "Hufflepuff"
    });
    expect(images[2].props).toEqual({
      src: "./images/Ravenclaw.png",
      alt: "Ravenclaw"
    });
    expect(images[3].props).toEqual({
      src: "./images/Slytherin.png",
      alt: "Slytherin"
    });
  });

  it("test display of house point", () => {
    const papers = testInstance.findAllByProps({
      className: "MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
    });
    expect(papers.length).toBe(4);
    for (let index = 0; index < 4; index++) {
      expect(papers[index].children.length).toBe(1);
      expect(papers[index].children[0]).toBe("0");
    }
  });

  it("test the two wizard lists (empty)", () => {
    const lists = testInstance.findAllByType("ul");
    expect(lists.length).toBe(2);
    expect(lists[0].children.length).toBe(0);
    expect(lists[1].children.length).toBe(0);
  });

  it("test the two adding form", () => {
    const forms = testInstance.findAllByProps({
      className: "MuiContainer-root MuiContainer-maxWidthLg"
    });
    expect(forms.length).toBe(4); //two add and two remove

    //ADD_PROFESSOR :
    let inputs = forms[0].children[0].children;
    expect(inputs.length).toBe(4);

    //inputs 0 : genre buttons
    let genreOptions = inputs[0].findAllByProps({
      className: "MuiButton-label"
    });
    expect(genreOptions.length).toBe(2); // M and F
    expect(genreOptions[0].children).toEqual(["M"]);
    expect(genreOptions[1].children).toEqual(["F"]);

    //inputs 1 : First Name
    let firstName = inputs[1].findAllByType("label");
    expect(firstName.length).toBe(1);
    expect(firstName[0].children).toEqual(["First Name"]);

    //inputs 2 : Last Name
    let lastName = inputs[2].findAllByType("label");
    expect(lastName.length).toBe(1);
    expect(lastName[0].children).toEqual(["Last Name"]);

    //inputs 3 : Add button
    let addButton = inputs[3].findAllByProps({
      className: "MuiButton-label"
    });
    expect(addButton.length).toBe(1);
    expect(addButton[0].children).toEqual(["Add"]);

    //ADD_STUDENT :
    inputs = forms[2].children[0].children;
    expect(inputs.length).toBe(5); //One more field than prof : house choice

    //inputs 0 : genre buttons
    genreOptions = inputs[0].findAllByProps({
      className: "MuiButton-label"
    });
    expect(genreOptions.length).toBe(2); // M and F
    expect(genreOptions[0].children).toEqual(["M"]);
    expect(genreOptions[1].children).toEqual(["F"]);

    //inputs 1 : First Name
    firstName = inputs[1].findAllByType("label");
    expect(firstName.length).toBe(1);
    expect(firstName[0].children).toEqual(["First Name"]);

    //inputs 2 : Last Name
    lastName = inputs[2].findAllByType("label");
    expect(lastName.length).toBe(1);
    expect(lastName[0].children).toEqual(["Last Name"]);

    //inputs 3 : House choice
    addButton = inputs[3].findAllByProps({
      className: "MuiButton-label"
    });
    expect(addButton.length).toBe(5);
    expect(addButton[0].children).toEqual(["Sorting Hat"]);
    expect(addButton[1].children).toEqual(["Gryffindor"]);
    expect(addButton[2].children).toEqual(["Hufflepuff"]);
    expect(addButton[3].children).toEqual(["Ravenclaw"]);
    expect(addButton[4].children).toEqual(["Slytherin"]);

    //inputs 4 : Add button
    addButton = inputs[4].findAllByProps({
      className: "MuiButton-label"
    });
    expect(addButton.length).toBe(1);
    expect(addButton[0].children).toEqual(["Add"]);

    //REMOVE_PROFESSOR
    inputs = forms[1].children;
    expect(inputs.length).toBe(1);
    let removeButton = inputs[0].findAllByProps({
      className: "MuiButton-label"
    });
    expect(removeButton.length).toBe(1);
    expect(removeButton[0].children).toEqual(["Remove Selected Wizard"]);

    //REMOVE_STUDENT
    inputs = forms[3].children;
    expect(inputs.length).toBe(1);
    removeButton = inputs[0].findAllByProps({
      className: "MuiButton-label"
    });
    expect(removeButton.length).toBe(1);
    expect(removeButton[0].children).toEqual(["Remove Selected Wizard"]);
  });
});
