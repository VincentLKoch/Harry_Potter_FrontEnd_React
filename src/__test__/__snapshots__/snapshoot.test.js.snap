// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshoot: Render correctly 1`] = `
Array [
  <div
    className="MuiContainer-root makeStyles-app-1 MuiContainer-maxWidthLg"
  >
    <div
      className="MuiBox-root MuiBox-root-13"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
        >
          <img
            alt="Gryffindor"
            src="./images/Gryffindor.png"
          />
          <div
            className="makeStyles-paper-117"
          >
            <div
              className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
            >
              0
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
        >
          <img
            alt="Hufflepuff"
            src="./images/Hufflepuff.png"
          />
          <div
            className="makeStyles-paper-117"
          >
            <div
              className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
            >
              0
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
        >
          <img
            alt="Ravenclaw"
            src="./images/Ravenclaw.png"
          />
          <div
            className="makeStyles-paper-117"
          >
            <div
              className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
            >
              0
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
        >
          <img
            alt="Slytherin"
            src="./images/Slytherin.png"
          />
          <div
            className="makeStyles-paper-117"
          >
            <div
              className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
            >
              0
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiContainer-root makeStyles-app-1 MuiContainer-maxWidthLg"
  >
    <div
      className="MuiBox-root MuiBox-root-146"
    >
      <div
        className="makeStyles-root-147"
      >
        <ul
          className="MuiGridList-root makeStyles-gridList-148"
          style={
            Object {
              "margin": -2,
            }
          }
        />
        <div
          className="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            style={
              Object {
                "margin": "10px",
              }
            }
          >
            <div
              aria-label="large contained secondary button group"
              className="MuiButtonGroup-root MuiButtonGroup-contained"
              role="group"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  M
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  F
                </span>
              </button>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink={true}
                htmlFor="First_Name"
                id="First_Name-label"
              >
                First Name
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="First_Name"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-268 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-269"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink={true}
                htmlFor="Last_Name"
                id="Last_Name-label"
              >
                Last Name
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="Last_Name"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-268 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-269"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Add
              </span>
            </button>
          </div>
        </div>
        <div
          className="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Remove Selected Wizard
            </span>
          </button>
        </div>
      </div>
      <div
        className="makeStyles-root-147"
      >
        <ul
          className="MuiGridList-root makeStyles-gridList-148"
          style={
            Object {
              "margin": -2,
            }
          }
        />
        <div
          className="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            style={
              Object {
                "margin": "10px",
              }
            }
          >
            <div
              aria-label="large contained secondary button group"
              className="MuiButtonGroup-root MuiButtonGroup-contained"
              role="group"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  M
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  F
                </span>
              </button>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink={true}
                htmlFor="First_Name"
                id="First_Name-label"
              >
                First Name
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="First_Name"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-268 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-269"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink={true}
                htmlFor="Last_Name"
                id="Last_Name-label"
              >
                Last Name
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="Last_Name"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-268 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-269"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              aria-label="large contained secondary button group"
              className="MuiButtonGroup-root MuiButtonGroup-contained"
              role="group"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Sorting Hat
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Gryffindor
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Hufflepuff
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Ravenclaw
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Slytherin
                </span>
              </button>
            </div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Add
              </span>
            </button>
          </div>
        </div>
        <div
          className="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Remove Selected Wizard
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
